core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -144,30 +141,30 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
     super.preStart()
     requests = Map.empty[TopicPartition, ActorRef]
     consumer = settings.createKafkaConsumer()
-    nextScheduledPoll = None
     commitsInProgress = 0
-    pollExpected = false
-    schedulePoll()
   }
 
   override def postStop(): Unit = {
-    nextScheduledPoll.foreach(_.cancel())
+    pollTask.cancel()
     consumer.close()
     super.postStop()
   }
 
   def poll() = {
-    if (pollExpected) {
-      pollExpected = false
-      nextScheduledPoll.foreach(_.cancel())
-      nextScheduledPoll = None
-
-      //set partitions to fetch
-      val partitionsToFetch = requests.keys.toSet
-      consumer.assignment().asScala.foreach { tp =>
-        if (partitionsToFetch.contains(tp)) consumer.resume(java.util.Collections.singleton(tp))
-        else consumer.pause(java.util.Collections.singleton(tp))
-      }
+    //set partitions to fetch
+    val partitionsToFetch = requests.keys.toSet
+    consumer.assignment().asScala.foreach { tp =>
+      if (partitionsToFetch.contains(tp)) consumer.resume(java.util.Collections.singleton(tp))
+      else consumer.pause(java.util.Collections.singleton(tp))
+    }
+
+    if (requests.isEmpty) {
+      // poll timeout of 0 doesn't work for the commits
+      val rawResult = consumer.poll(1)
```

Comment:
It's rather weird that the commits are not performed with `poll(0)`.
Is it safe to use `poll(1)` for minimum blocking? Seems to work with local Kafka. Could that be a problem when latency to Kafka server is larger?